server/index.js

```diff
@@ -12,6 +13,31 @@ app.use(staticMiddleware);
 
 app.use(express.json());
 
+app.post('/api/jokeApp/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      const sql = `
+  insert into "Users" ("username", "password")
+  values ($1, $2)
+  returning "usersId"
+  `;
+      const params = [username, hashedPassword];
+
+      db.query(sql, params)
+        .then(result => {
+          const [newRow] = result.rows;
+          res.json(newRow);
```

Comment:
This will work for a simple login and for the purposes of this app but for a true login you should be generating a webtoken then sending that back to the client. On the client side the token is what should be stored in localstorage, then with every request to your API the client should send the token and a middleware for the backend would verify auth on each request. This would also give you access to the user information including userId for every endpoint.